blockchain/chain.go

```diff
@@ -272,8 +272,47 @@ type BlockChain struct {
 	// so that the memory may be restored by the garbage collector if
 	// it is unlikely to be referenced in the future.
 	pruner *chainPruner
+
+	// The following maps are various caches for the stake version/voting
+	// system.  The goal of these is to reduce disk access to load blocks
+	// from disk.  Measurements indicate that it is slightly more expensive
+	// so setup the cache (<10%) vs doing a straight chain walk.  Every
+	// other subsequent call is >10x faster.  For example:
+	// stakeversion_test.go:930: running: shallow no cache with 20160 nodes
```

Comment:
I would prefer these results go into the commit message as opposed to the code.  The comment itself up to the "for example" is fine, and the results are definitely useful to document as part of the commit/PR, however, they are highly machine dependent, so I'd rather not put them in source code that will obviously be built and executed on a wide variety of hardware which makes them inherently incorrect.